lib/libthr/thread/thr_sig.c

```diff
@@ -296,13 +296,14 @@ handle_signal(struct sigaction *actp, int sig, siginfo_t *info, ucontext_t *ucp)
 		curthread->cancel_enable = 0;
 
 #ifdef CHERI_LIB_C18N
-	(void)sigfunc;
-#else
-	/* restore correct mask before calling user handler */
-	__sys_sigprocmask(SIG_SETMASK, &actp->sa_mask, NULL);
-
-	sigfunc = actp->sa_sigaction;
+	if (!_rtld_c18n_is_enabled())
```

Comment:
Cuddle brace?

Reply:
The `if` is surrounded by #ifdef's so the opening brace have to be outside the #ifdef. I don't know if there's a better way to write this.

Reply:
Hmm, you have two conditionals with only a comment in between.  I wonder if you could restructure this as:

```
#ifdef CHERI_LIB_C18N
     if (rtld_c18n_is_enabled()) {
         _rtld_siginvoke(sig, info, ucp, actp);
         goto invoked;
     }
#endif

     /* restore correct mask before calling user handler */
     ...
     
#ifdef CHER_LIB_C18N
invoked:
#endif
    err = errno
```

This will be a smaller diff less likely to conflict with upstream merges and it will be clearer what the alternate logic is.  Alternatively, you could indent all of that code into a giant `else` which I still think would be clearer than the current code.